cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -572,6 +600,17 @@ Result<std::shared_ptr<Schema>> ExtractSchemaToBind(const compute::Declaration&
   return bind_schema;
 }
 
+Result<std::unique_ptr<substrait::RelCommon>> GetRelCommonEmit(
```

Comment:
There is no need for this function to return a result (and then you can get rid of the move on the return value).